test/test_powerlaw_opacities.cpp

```diff
@@ -0,0 +1,323 @@
+        int n_wrong_h = 0;
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+        Kokkos::View<int, atomic_view> n_wrong_d("wrong");
+#else
+        PortableMDArray<int> n_wrong_d(&n_wrong_h, 1);
+#endif
+
+        portableFor(
+            "emissivities in funny units", 0, 100,
+            PORTABLE_LAMBDA(const int &i) {
+              Real jnu_funny = funny_units.EmissivityPerNuOmega(
+                  rho / rho_unit, temp / temp_unit, nu * time_unit);
+              Real jnu = opac.EmissivityPerNuOmega(rho, temp, nu);
+              if (FractionalDifference(jnu, jnu_funny * j_unit) > EPS_TEST) {
+                n_wrong_d() += 1;
+              }
+            });
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+        Kokkos::deep_copy(n_wrong_h, n_wrong_d);
+#endif
+        REQUIRE(n_wrong_h == 0);
+      }
+    }
+
+    THEN("We can fill an indexer allocated in a cell") {
+      int n_wrong_h = 0;
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+      Kokkos::View<int, atomic_view> n_wrong_d("wrong");
+#else
+      PortableMDArray<int> n_wrong_d(&n_wrong_h, 1);
+#endif
+      constexpr int nbins = 100;
+      constexpr int ntemps = 100;
+
+      constexpr Real lnu_min = 8;
+      constexpr Real lnu_max = 10;
+      Real nu_min = std::pow(10, lnu_min);
+      Real nu_max = std::pow(10, lnu_max);
+      Real dnu = (lnu_max - lnu_min) / (Real)(nbins - 1);
+
+      constexpr Real lt_min = 2;
+      constexpr Real lt_max = 4;
+      Real dt = (lt_max - lt_min) / (Real)(ntemps - 1);
+
+      Real *nu_bins = (Real *)PORTABLE_MALLOC(nbins * sizeof(Real));
+      Real *temp_bins = (Real *)PORTABLE_MALLOC(ntemps * sizeof(Real));
+      DataBox loglin_bins(Spiner::AllocationTarget::Device, ntemps, nbins);
+
+      portableFor(
+          "set nu bins", 0, nbins, PORTABLE_LAMBDA(const int &i) {
+            nu_bins[i] = std::pow(10, lnu_min + dnu * i);
+          });
+      portableFor(
+          "set temp bins", 0, ntemps, PORTABLE_LAMBDA(const int &i) {
+            temp_bins[i] = std::pow(10, lt_min + dt * i);
+          });
+
+      portableFor(
+          "Fill the indexers", 0, ntemps, PORTABLE_LAMBDA(const int &i) {
+            Real temp = temp_bins[i];
+            auto bins = loglin_bins.slice(i);
+            indexers::LogLinear J_log(bins, nu_min, nu_max, nbins);
+            opac.EmissivityPerNu(rho, temp, nu_bins, J_log, nbins);
+            Real Jtrue = opac.EmissivityPerNu(rho, temp, nu);
+            if (FractionalDifference(Jtrue, J_log(nu)) > EPS_TEST) {
+              n_wrong_d() += 1;
+            }
+          });
+
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+      Kokkos::deep_copy(n_wrong_h, n_wrong_d);
+#endif
+      REQUIRE(n_wrong_h == 0);
+
+      PORTABLE_FREE(nu_bins);
+      PORTABLE_FREE(temp_bins);
+      free(loglin_bins);
+    }
+
+    opac.Finalize();
+  }
+}
+
+TEST_CASE("Tophat Opacities", "[TopHat]") {
+  WHEN("We initialize a tophat neutrino opacity") {
+    neutrinos::Opacity opac = neutrinos::Tophat(1, 1e-2, 1e2);
+  }
+}
```

Comment:
I assume this is leftover from a copy-paste?

Reply:
Yep, removed :+1: